src/core/currency/wallet/currency-wallet-cleaners.ts

```diff
@@ -20,10 +33,10 @@ import { asJsonObject } from '../../../util/file-helpers'
  */
 export interface DiskMetadata {
   bizId?: number
-  category?: string
+  category?: string | null
   exchangeAmount: { [fiatCurrencyCode: string]: number }
-  name?: string
-  notes?: string
+  name?: string | null
+  notes?: string | null
```

Comment:
We don't actually want to store `null` on disk. The only reason we allow `null` in the user-facing type is to be able to delete fields. We need to revert this type definition, plus whatever changes are needed to the merging / updating code to make that work.

If could make sense to defined an `EdgeMetadataChange` type that includes `| null` for the fields that can be deleted, and revert the original `EdgeMetadata` type to how it was before. That makes it clear that metadata will never *contain* nulls. It's just a sigil for deleting fields.

Reply:
Understood. I noticed this when testing. I'd rather not intro a new type just yet but I can ensure we don't save null on disk.